Bukkit/src/main/java/com/plotsquared/bukkit/listener/EntityEventListener.java

```diff
@@ -310,16 +329,92 @@ public void onBigBoom(EntityExplodeEvent event) {
     @EventHandler(priority = EventPriority.HIGHEST, ignoreCancelled = true)
     public void onPeskyMobsChangeTheWorldLikeWTFEvent(EntityChangeBlockEvent event) {
         Entity e = event.getEntity();
-        if (!(e instanceof FallingBlock)) {
-            Location location = BukkitUtil.adapt(event.getBlock().getLocation());
-            PlotArea area = location.getPlotArea();
-            if (area != null) {
-                Plot plot = area.getOwnedPlot(location);
-                if (plot != null && !plot.getFlag(EntityChangeBlockFlag.class)) {
-                    plot.debug(e.getType() + " could not change block because entity-change-block = false");
+        Material type = event.getBlock().getType();
+        Location location = BukkitUtil.adapt(event.getBlock().getLocation());
+        if (!this.plotAreaManager.hasPlotArea(location.getWorldName())) {
```

Comment:
Removed this if-statement. By the way, I think I copy-pasted this from some other part of the code base, so it may be excessive in other places as well.